### PR TITLE
readme: Update readme file

### PR DESCRIPTION
**Pull request by oshoval**
Includes:
* Prerequisites.
* Info about how to deploy the plugin.
* Parameters.
* Info about how to develop the plugin.
* Basic manifests example.
* High level design section.

Beside that:
Ignore whitespace check on markdown files
(double space at the end means new line).

```release-note
None
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,127 @@
 # KubeSecondaryDNS
 DNS for KubeVirt VirtualMachines secondary interfaces
 
+## Prerequisites
+1. The KubeSecondaryDNS Deployment should be reachable from outside the cluster.
+2. The guest must have a guest agent installed, so the IP of the secondary interface will be reported.
+
```
Comment (dteplits) [inline README.md:7]: 3. If necessary - establish connectivity to cluster public IP via relevant DNS entity that is used to reach the authoritative KubeSecondaryDNS server, such as DNSResolver, TLD NameServer, etc.

File: README.md [hunk 2]
```diff
@@ -1,9 +1,127 @@
...
+## Parameters
+You can update those paramters in the manifest.
+
+`DOMAIN` (default: `""`) - Determines the suffix of the FQDN.  
+Can be used in order to differentiate between different clusters (by including the cluster name as part of it for example).  
+The FQDN would look as follows:  
+If `DOMAIN` is not empty `<interface_name>.<vm_name>.<namespace>.vm.<DOMAIN>`  
+If `DOMAIN` is empty `<interface_name>.<vm_name>.<namespace>.vm`  
+
+`NAME_SERVER_IP` (default: `""`) - TODO
```
Comment (dteplits) [inline README.md:40]: IP value that is defined and used in order to reach KubeSecondaryDNS server from outside the cluster.

Comment (oshoval) [reply README.md:40]: Thanks
Please rephrase it so it won't imply this IP is something we listen to, but focus about the NS and A records

Comment (oshoval) [reply README.md:40]: Lets rephrase it a bit, and add it on a follow PR, thanks

File: README.md (hunk 1 above)
Comment (oshoval) [reply README.md:11]: Added thanks

File: README.md [hunk 3]
```diff
@@ -1,9 +1,220 @@
...
 
+## Prerequisites
+1. The KubeSecondaryDNS Deployment should be reachable from outside the cluster.
+2. The guest must have a guest agent installed, so the IPs of the secondary interfaces will be reported.
+3. Kubevirt must be installed, else the plugin would exit with error after 2 minutes, due to
+the informer logic. It will be recreated according k8s logic.
+4. If necessary, establish connectivity to cluster public IP via a relevant DNS entity that is used to
+reach the authoritative KubeSecondaryDNS server, such as DNSResolver, TLD NameServer, etc.
+
+When running KubeSecondaryDNS on top on kubevirtci, we achieve [1] as follows:
```
Comment (AlonaKaplan) [inline README.md:12]: 1. on top of
2. Where is [1]?

File: README.md [hunk 4]
```diff
@@ -1,9 +1,220 @@
...
+1. Create a NodePort for the Deployment, use `31111` as the port.
```
Comment (AlonaKaplan) [inline README.md:13]: for the KubeSecondaryDNS Deployment

File: README.md [hunk 5]
```diff
@@ -1,9 +1,220 @@
...
+make create-nodeport
+```
+
+### Code related operations
+```shell
+# perform auto-formatting on the source code (if not done by your IDE)
+make fmt
+
+# run go vet
+make vet
```
Comment (AlonaKaplan) [inline README.md:99]: `make vendor` is missing

File: README.md (hunk 3 above)
Comment (oshoval) [reply README.md:12]: [1] few lines above

File: README.md (hunk 5 above)
Comment (oshoval) [reply README.md:99]: just selected few that looked more widely used, we don't include `make vendor` usually but i can add it

**Comment by oshoval** [comment]
Rebase

File: README.md [hunk 6]
```diff
@@ -1,9 +1,220 @@
...
+# run unit tests
+make test
+
+# bring up a local cluster with Kubernetes
+make cluster-up
+
+# build and deploy KubeSecondaryDNS
+make cluster-sync
+
+# run e2e tests on the cluster, requires cluster with installed operator,
```
[the remark this answers is not in the document]
Comment (oshoval) [reply README.md:59]: replaced with KubeSecondaryDNS

**Comment by oshoval** [comment]
Addressed comments

**Comment by oshoval** [comment]
Fixed whitespace to ignore markdown files

File: README.md [hunk 7]
```diff
@@ -1,9 +1,231 @@
 # KubeSecondaryDNS
 DNS for KubeVirt VirtualMachines secondary interfaces
 
+## Prerequisites
+1. The KubeSecondaryDNS Deployment should be reachable from outside the cluster.  
+It can be exposed using NodePort, Load Balancer, Ingress or any other methodology.  
+The IP to reach the KubeSecondaryDNS for outside the cluster would be called from now on
```
Comment (AlonaKaplan) [inline README.md:7]: s/for/from

File: README.md [hunk 8]
```diff
@@ -1,9 +1,231 @@
...
+      - disk:
+          bus: virtio
+        name: containerdisk
+      - disk:
+          bus: virtio
+        name: cloudinitdisk
+      interfaces:
+      - bridge: {}
+        name: nic1
+    features:
```
Comment (AlonaKaplan) [inline README.md:189]: I believe `features` and `machine` can be dropped.

File: README.md [hunk 9]
```diff
@@ -1,9 +1,231 @@
...
+StatusMonitor is watching VMI status changes via k8s API,
+and sending the info to the ZoneManager about the secondary interfaces.
+The ZoneManager builds the Zone File that the CoreDNS expects according to the received status.
+
+The Zone File consists of:
+1. SOA header (DNS parameters)
+2. List of FQDN and associated IP addresses.
+
+ZoneManager updates the `/zones` folder which is shared between the containers, with the updated data.  
+CoreDNS monitors the `/zones` folder, and updates its db accordingly (using CoreDNS `auto` plugin).
```
Comment (AlonaKaplan) [inline README.md:230]: s/db/DB

**Comment by oshoval** [comment]
Addressed comments (beside https://github.com/kubevirt/kubesecondarydns/pull/37/files#r1045608648 yet)

**Comment by oshoval** [comment]
Updated manifest, also updated the image

File: README.md (hunk 2 above)
Comment (dteplits) [reply README.md]: The @ symbol in the line before is responsible to add the $ORIGIN value (=domain name) before each one of those two lines.
Actually, I think I miss there a nameserver prefix in the A line, I'll add it in a separate PR.
Thanks for noticing.

Comment (oshoval) [reply README.md]: Thanks, added ns to the A line

**Comment by oshoval** [comment]
Rebase 
Address this https://github.com/kubevirt/kubesecondarydns/pull/37#discussion_r1048529963

**Comment by AlonaKaplan** [comment]
/lgtm
/approve
Thanks!

**Pull request merged**